### PR TITLE
Added fuel vaporization to fluid puddles.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
Adding some very basic fire interaction to fuel puddles, as a stopover between now and a proper fire rewrite. Essentially just causes puddles to vaporize any fuel in their reagents on exposure to sufficient heat, meaning that zone fires will start.
DNM, have not tested locally due to putting it together at work.

### CODE REVIEW

**Comment by MistakeNot4892** [comment]
Does not work :(

**Comment by MistakeNot4892** [comment]
Working to an extent - not super reliable and seems to make welding tanks more dangerous than hydrogen tanks, but it does at least make fuel spills dangerous.

**Review by afterthought2 — APPROVED** [review]

File: code/modules/ZAS/Fire.dm [hunk 1]
```diff
@@ -125,6 +130,12 @@ If it gains pressure too slowly, it may leak or just rupture instead of explodin
 	for(var/atom/A in loc)
 		A.fire_act(air_contents, air_contents.temperature, air_contents.volume)
 
+	// prioritize nearby fuel overlays first
+	for(var/direction in global.cardinal)
```
Comment (MistakeNot4892) [inline code/modules/ZAS/Fire.dm:134]: DNM, this is going to stack overflow.

Comment (MistakeNot4892) [reply code/modules/ZAS/Fire.dm:134]: Isn't an issue, got confused.

**Pull request merged**